### ISSUE TITLE
Shouldn't call attachShadow if the element already has a shadow DOM

### ISSUE THREAD

**Issue by matthewp** [skatejs/skatejs#1140]
I have a custom element that creates shadow roots on behalf of other elements. When one of those other custom elements is a Skate component it will throw because of [this line](https://github.com/skatejs/skatejs/blob/b5b7097b2f2fd5c154fab56841876aae4c87c23a/src/with-render.js#L13).

I think that it should check if `this.shadowRoot` exists already and prefer that over calling attachShadow. Or create a attachShadowIfNeeded function.

Thoughts @treshugart @Hotell ?

**Comment by matthewp** [comment]
Changing that line to:

```js
this[_shadowRoot] = this[_shadowRoot] || (this[_shadowRoot] = (this.shadowRoot || attachShadow(this)));
```

Would fix it.

**Comment by matthewp** [comment]
Closed by #1141
